static/js/src/advantage/subscribe/react/APICalls/RegisterPaymentMethod.jsx

```diff
@@ -0,0 +1,93 @@
+import { CardElement, useElements, useStripe } from "@stripe/react-stripe-js";
+import { useMutation } from "react-query";
+import {
+  ensurePurchaseAccount,
+  postCustomerInfoToStripeAccount,
+} from "../../../contracts-api";
+
+const registerPaymentMethod = () => {
+  const stripe = useStripe();
+  const elements = useElements();
+
+  const mutation = useMutation(async (formData) => {
+    const {
+      name,
+      organisation,
+      email,
+      address,
+      city,
+      country,
+      postalCode,
+      usState,
+      CAProvince,
+      VATNumber,
+    } = formData;
+
+    const card = elements.getElement(CardElement);
+
+    const addressObject = {
+      city: city,
+      country: country,
+      line1: address,
+      postal_code: postalCode,
+      state: country === "US" ? usState : CAProvince,
+    };
+
+    const { error, paymentMethod } = await stripe.createPaymentMethod({
+      type: "card",
+      card: card,
+      billing_details: {
+        name: name,
+        email: email,
+        address: addressObject,
+      },
+    });
+
+    if (error) {
+      throw new Error(error.code);
+    }
+
+    const accountRes = await ensurePurchaseAccount({
+      email: email,
+      accountName: organisation || name,
+      paymentMethodID: paymentMethod.id,
+      country,
+    });
+
+    if (accountRes.code) {
+      if (accountRes.code === "unauthorized") {
+        throw new Error("email_already_exists");
+      } else {
+        throw new Error(
+          JSON.parse(accountRes.message)?.decline_code || accountRes.code
+        );
+      }
+    }
+
+    const customerInfoRes = await postCustomerInfoToStripeAccount({
+      paymentMethodID: paymentMethod.id,
+      accountID: accountRes.accountID,
+      address: addressObject,
+      name: name,
+      taxID: {
+        type: "eu_vat",
+        value: VATNumber,
+      },
+    });
+
+    if (customerInfoRes.errors) {
+      //We ignore VAT errors but throw the others
+      if (JSON.parse(customerInfoRes.errors).code !== "tax_id_invalid")
+        throw new Error(JSON.parse(customerInfoRes.errors).code);
```

Comment:
Where has the curly brackets gone?

Reply:
I'm surprised the linter allowed this - might be best to add that rule to the linter.

Reply:
I'm writing Python now! :smile: 
I think the linter is fine with it for one liners. I'll add them back in for consistency.